File: frontend/src/app/enterprise/video/model/dto/VideoCardDto.ts
export class VideoLabelDto {
  Cod: string = '';
  Name: string = '';
}

export class VideoCardDto {
  VideoCod: string = '';
  Title: string = '';
  ShortDescription: string = '';
  ThumbnailUrl: string = '';
  SourceType: string = '';
  Duration: string = '';
  FileSizeBytes: number = 0;
  FileSizeLabel: string = '';
  ResolutionWidth: number = 0;
  ResolutionHeight: number = 0;
  ResolutionLabel: string = '';
  ViewCount: number = 0;
  PublishDate: string = '';
  CreationDate: string = '';
  PrimaryCategoryCod: string = '';
  PrimaryCategoryName: string = '';
  Categories: VideoLabelDto[] = [];
  Actors: VideoLabelDto[] = [];
  Tags: VideoLabelDto[] = [];
}

File: desktop/src/main/java/com/ccadmin/app/video/model/dto/VideoCardDto.java
package com.ccadmin.app.video.model.dto;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class VideoCardDto {
    public String VideoCod;
    public String Title;
    public String ShortDescription;
    public String ThumbnailUrl;
    public String SourceType;
    public String Duration;
    public Long FileSizeBytes;
    public String FileSizeLabel;
    public Integer ResolutionWidth;
    public Integer ResolutionHeight;
    public String ResolutionLabel;
    public Long ViewCount;
    public LocalDateTime PublishDate;
    public LocalDateTime CreationDate;
    public String PrimaryCategoryCod;
    public String PrimaryCategoryName;
    public List<VideoLabelDto> Categories = new ArrayList<>();
    public List<VideoLabelDto> Actors = new ArrayList<>();
    public List<VideoLabelDto> Tags = new ArrayList<>();
}
